docs/resources/scenarios/grade-calculator.md

```diff
@@ -31,82 +32,150 @@ You'll develop a script that totals the grades for each point category. It will
+      // Get the worksheet and validate the data.
+      let studentsRange = workbook.getActiveWorksheet().getUsedRange();
+      if (studentsRange.getColumnCount() !== 6) {
+        throw new Error(`The required columns are not present. Expected headers: "Student ID | Assignment score | Mid-term | Final | Total | Grade" columns.`);
+      }
+
+      let studentData = studentsRange.getValues();
+
+      // Clear the total and grade columns.
+      studentsRange.getColumn(4).getCell(1, 0).getAbsoluteResizedRange(studentData.length - 1, 2).clear();
+
+      // Clear all conditional formatting.
+      workbook.getActiveWorksheet().getUsedRange().clearAllConditionalFormats();
+
+      // Use regular expressions to read the max score from the assignment, mid-term, and final score columns.
+      let maxScores: string[] = [];
+      const assignmentMaxMatches = studentData[0][1].match(/\d+/);
+      const midtermMaxMatches = studentData[0][2].match(/\d+/);
+      const finalMaxMatches = studentData[0][3].match(/\d+/);
+
+      // Check the matches happened before proceeding.
+      if (!(assignmentMaxMatches && midtermMaxMatches && finalMaxMatches)) {
+        throw new Error(`The scores are not present in the column headers. Expected format: "|Assignments (n)|Mid-term (n)|Final (n)"`);
+      }
+
+      // Use the first (and only) match from the regular expressions as the max scores.
+      maxScores = [assignmentMaxMatches[0], midtermMaxMatches[0], finalMaxMatches[0]];
+
+      // Set conditional formatting for each of the assignment, mid-term and final scores columns.
+      maxScores.forEach((score, i) => {
+        let range = studentsRange.getColumn(i + 1).getCell(0, 0).getRowsBelow(studentData.length - 1);
+        setCellValueConditionalFormatting(
+          score,
+          range,
+          "#9C0006",
+          "#FFC7CE",
+          ExcelScript.ConditionalCellValueOperator.greaterThan
+        )
+      });
+
+      // Store the current range information to avoid calling the workbook in the loop.
+      let studentsRangeFormulas = studentsRange.getColumn(4).getFormulasR1C1();
+      let studentsRangeValues = studentsRange.getColumn(5).getValues();
+
+      /* Iterate over each of the student rows and compute the total score and letter grade.
+      * Note that iterator starts at index 1 (skip first row).
+      */
+      for (let i = 1; i < studentData.length; i++) {
+        // If any of the scores are invalid, skip processing it.
+        if (studentData[i][1] > maxScores[0] ||
+          studentData[i][2] > maxScores[1] ||
+          studentData[i][3] > maxScores[2]) {
+          continue;
         }
-        if (row[3] < 0 || row[3] > finalsMax) {
-          studentsRange.getCell(i, 3).format.fill.color = "Red";
-          valid = false;
+        const total = studentData[i][1] + studentData[i][2] + studentData[i][3];
+        let grade: string;
+        switch (true) {
+          case total < 60:
+            grade = "E";
```

Comment:
Not sure about this... If you accept this, then change subsequent references
```suggestion
            grade = "F";
```

Reply:
Sudhi told me a lot of schools use 'E' instead of 'F' now, but you're right that 'F' is more common. I'll change it.